auth/src/main/java/io/grpc/auth/ClientAuthInterceptor.java

```diff
@@ -96,11 +104,29 @@ protected void checkedStart(Listener<RespT> responseListener, Metadata headers)
     };
   }
 
-  private Map<String, List<String>> getRequestMetadata() {
+  private URI serviceUri(Channel channel, MethodDescriptor<?, ?> method) throws StatusException {
+    String authority = channel.authority();
+    if (authority == null) {
+      throw Status.UNAUTHENTICATED.withDescription("Channel has no authority").asException();
+    }
+    final String suffixToStrip = ":443";
+    if (authority.endsWith(suffixToStrip)) {
+      authority = authority.substring(0, authority.length() - suffixToStrip.length());
```

Comment:
For readability, could you do all of this in a method `String stripDefaultPort(String authority)`?